packages/api/lib/writeRecords/write-granules.js

```diff
@@ -176,6 +177,103 @@ const _writePostgresGranuleViaTransaction = async ({
   return granuleCumulusId;
 };
 
+/**
+ * Update granule status
+ *
+ * @param {Object} params
+ * @param {number} params.granuleCumulusId - Cumulus ID of a granule
+ * @param {string} params.granuleId - granule ID from the payload
+ * @param {Knex} params.knex - knex Client
+ * @param {string} params.status - new status for granule
+ * @returns {Promise}
+ * @throws {Error}
+ */
+async function updateGranuleStatus({
+  granule,
+  knex,
+  status,
+  error = '',
+  errorCause = '',
+  granuleModel = new Granule(),
+  granulePgModel = new GranulePgModel(),
+  collectionPgModel = new CollectionPgModel(),
+}) {
+  const { granuleId, collectionId } = granule;
+  const errorObject = error && errorCause
+    ? {
+      Error: error,
+      Cause: errorCause,
+    }
+    : {};
+  try {
+    const collectionCumulusId = await collectionPgModel.getRecordCumulusId(
+      knex,
+      deconstructCollectionId(collectionId)
+    );
+    const granuleCumulusId = await granulePgModel.getRecordCumulusId(
+      knex,
+      {
+        granule_id: granuleId,
+        collection_cumulus_id: collectionCumulusId,
+      }
+    );
+
+    await createRejectableTransaction(knex, async (trx) => {
+      await granulePgModel.update(
+        trx,
+        { cumulus_id: granuleCumulusId },
+        {
+          status,
+          error: errorObject,
+        }
+      );
+      // delete the execution field as well
+      await granuleModel.update(
+        { granuleId },
+        {
+          status,
+          error: errorObject,
+        },
+        ['execution']
```

Comment:
So this is relevant in the case of writing granules as "queued", but i don't think it is relevant if we're writing them as "failed". When we're updating a granule as "queued", it's going to have an execution, so we want to delete the `execution` value in Dynamo if there is one. But when updating a granule as "failed", we want to preserve whatever value there is for `execution`. 

This function should probably just accept an optional array parameter named `apiGranuleDeleteFields` which could be set to `['execution']` in `updateGranuleStatusToQueued`

Reply:
ah. got it. making that change now